endpoints/openrtb2/auction_test.go

```diff
@@ -120,196 +191,178 @@ func TestExplicitUserId(t *testing.T) {
+		&config.Configuration{
+			MaxRequestSize:     maxSize,
+			BlacklistedApps:    test.Config.BlacklistedAppArr,
+			BlacklistedAppMap:  test.Config.blacklistedAppMap,
+			BlacklistedAccts:   test.Config.BlacklistedAccountArr,
+			BlacklistedAcctMap: test.Config.blacklistedAccountMap,
+			AccountRequired:    test.Config.AccountReq,
+		},
+		theMetrics,
+		analyticsConf.NewPBSAnalytics(&config.Analytics{}),
+		disabledBidders,
+		[]byte(test.Config.AliasJSON),
+		bidderMap,
+	)
+
+	request := httptest.NewRequest("POST", "/openrtb2/auction", bytes.NewReader(test.BidRequest))
+	recorder := httptest.NewRecorder()
+	endpoint(recorder, request, nil) //Request comes from the unmarshalled mockBidRequest
+	return recorder.Code, recorder.Body.String()
 }
 
-// TestAliasedRequests makes sure we handle (default) aliased bidders properly
-func TestAliasedRequests(t *testing.T) {
-	tests := &getResponseFromDirectory{
-		dir:           "sample-requests/aliased",
-		payloadGetter: noop,
-		messageGetter: nilReturner,
-		expectedCode:  http.StatusOK,
-		aliased:       true,
+func jsonParseTestFile(t *testing.T, fileData []byte, testFile string) *testCaseData {
+	t.Helper()
+
+	parsedTestData := &testCaseData{}
+	var err error
+
+	// Get testCaseData values
+	parsedTestData.BidRequest, _, _, err = jsonparser.Get(fileData, "mockBidRequest")
+	assert.NoError(t, err, "Error jsonparsing root.mockBidRequest from file %s. Desc: %v.", testFile, err)
+
+	parsedReturnCode, err := jsonparser.GetInt(fileData, "expectedReturnCode")
+	assert.NoError(t, err, "Error jsonparsing root.code from file %s. Desc: %v.", testFile, err)
+
+	parsedTestData.ExpectedReturnCode = int(parsedReturnCode)
+
+	if parsedTestData.ExpectedReturnCode != 200 {
+		// Get expected error, fail if parsing error
+		parsedTestData.ExpectedErrorMessage, err = jsonparser.GetString(fileData, "expectedErrorMessage")
+		assert.NoError(t, err, "Error jsonparsing root.expectedErrorMessage from file %s. Desc: %v.", testFile, err)
+	} else {
+		// Get expected response, fail if parsing error
+		parsedTestData.ExpectedBidResponse, _, _, err = jsonparser.Get(fileData, "expectedBidResponse")
+		assert.NoError(t, err, "Error jsonparsing root.expectedBidResponse from file %s. Desc: %v.", testFile, err)
 	}
-	tests.assert(t)
-}
 
-// TestDisabledBidders makes sure we don't break when encountering a disabled bidder
-func TestDisabledBidders(t *testing.T) {
-	badTests := &getResponseFromDirectory{
-		dir:             "sample-requests/disabled/bad",
-		payloadGetter:   getRequestPayload,
-		messageGetter:   getMessage,
-		expectedCode:    http.StatusBadRequest,
-		aliased:         false,
-		disabledBidders: []string{"appnexus", "rubicon"},
-		adaptersConfig: map[string]config.Adapter{
-			"appnexus": {Disabled: true},
-			"rubicon":  {Disabled: true},
-		},
+	// Get testCaseConfig values
+	parsedTestData.Config = &testConfigValues{}
+	accReq, err := jsonparser.GetBoolean(fileData, "mockConfig", "accountRequired")
```

Comment:
The term mock here is a little strange IMHO. I would recommend just calling this the 'config' section of the json file.